src/index.ts

```diff
@@ -139,7 +139,7 @@ export async function retry<T> (
 
     const onError = async (err: any) => {
       if (options.handleError) {
-        options.handleError(err, context, options);
+        await options.handleError(err, context, options);
```

Comment:
A test that ensures the handle error completes prior to another attempt occurring would be great.

Reply:
Good suggestion. Just pushed up a test.